sensor_msgs_py/sensor_msgs_py/point_cloud2.py

```diff
@@ -157,28 +228,52 @@ def create_cloud(header, fields, points):
                    the fields parameter)
     :return: The point cloud as sensor_msgs.msg.PointCloud2
     """
-    cloud_struct = struct.Struct(_get_struct_fmt(False, fields))
-
-    buff = ctypes.create_string_buffer(cloud_struct.size * len(points))
-
-    point_step, pack_into = cloud_struct.size, cloud_struct.pack_into
-    offset = 0
-    for p in points:
-        pack_into(buff, offset, *p)
-        offset += point_step
-
-    return PointCloud2(header=header,
-                       height=1,
-                       width=len(points),
-                       is_dense=False,
-                       is_bigendian=False,
-                       fields=fields,
-                       point_step=cloud_struct.size,
-                       row_step=cloud_struct.size * len(points),
-                       data=buff.raw)
-
-
-def create_cloud_xyz32(header, points):
+    # Check if input is numpy array
+    if isinstance(points, np.ndarray):
+        # Check if this is an unstructured array
+        if points.dtype.names is None:
+            assert all(fields[0].datatype == field.datatype for field in fields[1:]), \
+                "All fields need to have the same datatype. Pass a structured NumPy array \
+                    with multiple dtypes otherwise."
+            # Convert unstructured to structured array
+            points = unstructured_to_structured(
+                points,
+                dtype=dtype_from_fields(fields))
+        else:
+            assert points.dtype == dtype_from_fields(fields), \
+                "PointFields and structured NumPy array dtype do not match for all fields! \
+                    Check their field order, names and types."
+    else:
+        # Cast python objects to structured NumPy array (slow)
+        points = np.array(
+            # Points need to be tuples in the structured array
+            list(map(tuple, points)),
+            dtype=dtype_from_fields(fields))
+
+    # Handle organized clouds
+    assert len(points.shape) <= 2, \
+        "Too many dimensions for organized cloud! \
+            Points can only be organized in max. two dimensional space"
+    height = 1
+    width = points.shape[0]
+    # Check if input points are an organized cloud (2D array of points)
+    if len(points.shape) == 2:
+        height = points.shape[1]
+
+    # Put everything together
+    return PointCloud2(
+        header=header,
+        height=height,
+        width=width,
+        is_dense=False,
+        is_bigendian=sys.byteorder != 'little',
+        fields=fields,
+        point_step=points.dtype.itemsize,
+        row_step=(points.dtype.itemsize * width),
+        data=points.tobytes())
```

Comment:
@Flova I just spent a while improving performance for PointCloud2 message creation on the ros2_numpy repo, and thought I'd pass on some knowledge. I noticed that  doing `data=np_array.tobytes()` can be quite slow on large point clouds. I have some code in [this PR](https://github.com/Box-Robotics/ros2_numpy/pull/3) using a memoryview and `array.array.frombytes()` which I found to be 100x faster.

Let me know what you think- there's some overlap in what we're working on I think.